sklearn/metrics/tests/test_ranking.py

```diff
@@ -418,13 +418,13 @@ def test_roc_curve_drop_intermediate():
     y_true = [0, 0, 0, 0, 1, 1]
     y_score = [0.0, 0.2, 0.5, 0.6, 0.7, 1.0]
     tpr, fpr, thresholds = roc_curve(y_true, y_score, drop_intermediate=True)
-    assert_array_almost_equal(thresholds, [2.0, 1.0, 0.7, 0.0])
+    assert_array_almost_equal(thresholds, [1.0, 1.0, 0.7, 0.0])
```

Comment:
These changes here annoyed me because, for the same thresholds, we can now have to point on the ROC curve. It is also a little weird.

Reply:
This also bothers me, but if `tpr[0] == tpr[1]` and `fpr[0] == fpr[1]` this might be okay (?). Is that the case?

Reply:
> These changes here annoyed me because, for the same thresholds, we can now have two points on the ROC curve. It is also a little weird.

We could insert the extra threshold only when needed as follows:

```python
if min_threshold >= 0 and max_threshold <= 1:
    # For probability scores, insert an extra point only when the 1.0 boundary 
    # is not reached.
    if max_threshold < 1:
        tps = np.r_[0, tps]
        fps = np.r_[0, fps]
        thresholds = np.r_[1.0, thresholds]
else:
    # For unbounded decision_function thresholds, insert an arbitrary offset
    # max threshold.
    tps = np.r_[0, tps]
    fps = np.r_[0, fps]
    thresholds = np.r_[max_threshold + 1.0, thresholds]
```

Reply:
I was thinking about doing this. In some way, it means that there is one case where one can never have the (0, 0) case because a probability of one will not reach the starting point. While this is a change, it feel the right thing to do.

Reply:
So this is even more complex. The computation of the `roc_auc` will expect to have at least 2 points on the ROC curve so we must add a point (0, 0) for some corner cases where the threshold could already be `1.0`.

As stated by @vitaliset, if we want to respect the rule `if .predict_proba > threshold, then predict 1; otherwise, predict 0` then the only way that I see is to add a `1.0 + eps` when `thresholds[0]` is already 1.0.

So this comeback to the original version with less arbitrary behaviour. We can also publicly document this case.